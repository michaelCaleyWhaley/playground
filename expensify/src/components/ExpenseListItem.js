import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ description, amount, createdAt, note, id }) => (
  <li>
    <Link to={`/edit/${id}`}>
      <h3>Description: {description}</h3>
    </Link>
    <p>Cost: Â£{amount}</p>
    <p>CreatedAt: {createdAt}</p>
    <p>Note: {note}</p>
  </li>
);

export default ExpenseListItem;
