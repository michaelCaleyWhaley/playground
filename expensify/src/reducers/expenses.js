// expenses reducer
// default state
const expensesReducerDefaultState = [];
const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];
    case "REMOVE_EXPENSE":
      return state.filter(expense => expense.id !== action.id);
      return state;
    case "EDIT_EXPENSE":
      return state.map(currentValue => {
        if (currentValue.id === action.id) {
          return {
            ...currentValue,
            ...action.updates,
          };
        } else {
          return currentValue;
        }
      });
    default:
      return state;
  }
};

export default expensesReducer;
