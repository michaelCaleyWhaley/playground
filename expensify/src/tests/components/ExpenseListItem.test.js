import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("should render ExpenseListItem correctly", () => {
  const props = expenses[0];
  const wrapper = shallow(<ExpenseListItem {...props} />);
  expect(wrapper).toMatchSnapshot();
});
