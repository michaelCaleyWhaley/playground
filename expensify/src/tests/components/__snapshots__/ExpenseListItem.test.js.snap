// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<li>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Description: 
      Gum
    </h3>
  </Link>
  <p>
    Cost: Â£
    195
  </p>
  <p>
    CreatedAt: 
  </p>
  <p>
    Note: 
  </p>
</li>
`;
